[2025-07-27T19:26:52.408+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2025-07-27T19:26:52.445+0000] {taskinstance.py:2073} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: training_pipeline.process_data manual__2025-07-27T19:26:39.125440+00:00 [queued]>
[2025-07-27T19:26:52.455+0000] {taskinstance.py:2073} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: training_pipeline.process_data manual__2025-07-27T19:26:39.125440+00:00 [queued]>
[2025-07-27T19:26:52.483+0000] {taskinstance.py:2303} INFO - Starting attempt 1 of 4
[2025-07-27T19:26:52.498+0000] {taskinstance.py:2327} INFO - Executing <Task(PythonOperator): process_data> on 2025-07-27 19:26:39.125440+00:00
[2025-07-27T19:26:52.514+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'training_pipeline', 'process_data', 'manual__2025-07-27T19:26:39.125440+00:00', '--job-id', '4', '--raw', '--subdir', 'DAGS_FOLDER/training_pipeline.py', '--cfg-path', '/tmp/tmpg50ydb9c']
[2025-07-27T19:26:52.515+0000] {logging_mixin.py:188} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:61 DeprecationWarning: This process (pid=347) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2025-07-27T19:26:52.517+0000] {standard_task_runner.py:91} INFO - Job 4: Subtask process_data
[2025-07-27T19:26:52.518+0000] {standard_task_runner.py:63} INFO - Started process 364 to run task
[2025-07-27T19:26:52.533+0000] {logging_mixin.py:188} WARNING - /home/***/.local/lib/python3.12/site-packages/***/settings.py:195 DeprecationWarning: The sql_alchemy_conn option in [core] has been moved to the sql_alchemy_conn option in [database] - the old setting has been used, but please update your config.
[2025-07-27T19:26:52.571+0000] {task_command.py:426} INFO - Running <TaskInstance: training_pipeline.process_data manual__2025-07-27T19:26:39.125440+00:00 [running]> on host d1e83e43e29f
[2025-07-27T19:26:52.707+0000] {taskinstance.py:2644} INFO - Exporting env vars: AIRFLOW_CTX_DAG_EMAIL='mlops_team@example.com' AIRFLOW_CTX_DAG_OWNER='mlops' AIRFLOW_CTX_DAG_ID='training_pipeline' AIRFLOW_CTX_TASK_ID='process_data' AIRFLOW_CTX_EXECUTION_DATE='2025-07-27T19:26:39.125440+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-07-27T19:26:39.125440+00:00'
[2025-07-27T19:26:52.709+0000] {taskinstance.py:430} INFO - ::endgroup::
[2025-07-27T19:26:52.786+0000] {data_processing.py:40} INFO - Loaded 288 records from S3: data/historical_BTC_USD_20250727192648.csv
[2025-07-27T19:26:52.790+0000] {data_processing.py:84} INFO - Combined 288 records for BTC-USD
[2025-07-27T19:26:52.840+0000] {data_processing.py:162} INFO - Removed 50 rows with NaN values. Final shape: (238, 84)
[2025-07-27T19:26:52.843+0000] {logging_mixin.py:188} WARNING - /opt/***/src/data_processing.py:230 FutureWarning: DataFrame.fillna with 'method' is deprecated and will raise in a future version. Use obj.ffill() or obj.bfill() instead.
[2025-07-27T19:26:52.847+0000] {logging_mixin.py:188} WARNING - /opt/***/src/data_processing.py:177 FutureWarning: DataFrame.fillna with 'method' is deprecated and will raise in a future version. Use obj.ffill() or obj.bfill() instead.
[2025-07-27T19:26:52.849+0000] {data_processing.py:179} INFO - Prepared features: (238, 80), Target: (238,)
[2025-07-27T19:26:52.850+0000] {data_processing.py:180} INFO - Feature columns: 80
[2025-07-27T19:26:53.231+0000] {data_processing.py:202} INFO - Saved preprocessor to S3: s3://crypto-mlops-bucket/models/preprocessors/scaler_BTC_USD.pickle
[2025-07-27T19:26:53.233+0000] {python.py:237} INFO - Done. Returned value was: None
[2025-07-27T19:26:53.234+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2025-07-27T19:26:53.279+0000] {taskinstance.py:1205} INFO - Marking task as SUCCESS. dag_id=training_pipeline, task_id=process_data, execution_date=20250727T192639, start_date=20250727T192652, end_date=20250727T192653
[2025-07-27T19:26:53.327+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 0
[2025-07-27T19:26:53.354+0000] {taskinstance.py:3482} INFO - 1 downstream tasks scheduled from follow-on schedule check
[2025-07-27T19:26:53.391+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
